cmd/testing-server/main.go

```diff
@@ -47,30 +54,102 @@ func createDeviceHandler(w http.ResponseWriter, r *http.Request) {
 		}
 		log.Printf("Creating device " + deviceName + " and group-name " + groupnames[0] + " on server " + apiServer)
 		cmd := exec.Command("./device-register",
-												"--name",
-												deviceName,
-												"--group",
-												groupnames[0],
-												"--password",
-												"password_" + deviceName,
-												"--api",
-												apiServer,
-												"--ignore-minion-id",
-												"--remove-device-config")
+			"--name",
+			deviceName,
+			"--group",
+			groupnames[0],
+			"--password",
+			"password_"+deviceName,
+			"--api",
+			apiServer,
+			"--ignore-minion-id",
+			"--remove-device-config")
 
 		cmd.Dir = "/code/device-register"
 
 		if output, err := cmd.CombinedOutput(); err != nil {
 			outputString := string(output)
-			log.Printf("Error was " + outputString)
+			log.Printf("Error was %v", outputString)
 			http.Error(w, outputString, http.StatusInternalServerError)
 		} else {
 			log.Printf("device created")
-			io.WriteString(w, "Success")
+			restartThermalUploader()
+			deviceID, err := getDeviceID()
+			if err != nil {
+				http.Error(w, fmt.Sprintf("Could not read device id %v", err), http.StatusInternalServerError)
+			}
+			io.WriteString(w, fmt.Sprintf("%d", deviceID))
```

Comment:
It needs to "Sucess" in the browser or the error.   fmt.Sprintf("Success.   Device id is %d", deviceId)).